version: '3.8'

services:
  postgres:
    image: postgres:12.17-alpine3.19
    restart: unless-stopped
    environment:
      - POSTGRES_USER=root
      - POSTGRES_PASSWORD=root
      - POSTGRES_DB=pizza
    ports:
      - '5432:5432'
    networks:
    - backend
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U root -d pizza"]
      interval: 5s
      timeout: 10s
      retries: 20
    volumes:
      - db_data:/var/lib/postgresql/data

  node: 
    build: 
      context: ./node
      dockerfile: Dockerfile.dev
    ports:
      - "3000:3000"
    depends_on:
      - postgres
    links:
      - postgres
    networks:
      - backend
      - frontend
    volumes:
      - ./node:/home/node

  react:
    build: ./frontend
    ports:
      - "8000:8000"
    depends_on:
      - node
    links:
      - node
    networks:
      - frontend
    volumes:
      - ./frontend:/home/react

volumes:
  db_data:

networks:
  backend:
  frontend:
